gqcp/include/QCModel/HF/UHF.hpp

```diff
@@ -376,22 +376,14 @@ class UHF {
         const auto& orbital_space_sigma_bar = orbital_space.component(sigma_bar);
 
         // Determine the number of occupied and virtual orbitals for both spin components.
-        const auto n_occ_sigma = orbital_space_sigma.numberOfOrbitals(OccupationType::k_occupied);
-        const auto n_virt_sigma = orbital_space_sigma.numberOfOrbitals(OccupationType::k_virtual);
+        const auto& n_occ_sigma = orbital_space_sigma.numberOfOrbitals(OccupationType::k_occupied);
+        const auto& n_virt_sigma = orbital_space_sigma.numberOfOrbitals(OccupationType::k_virtual);
 
-        const auto n_occ_sigma_bar = orbital_space_sigma_bar.numberOfOrbitals(OccupationType::k_occupied);
-        const auto n_virt_sigma_bar = orbital_space_sigma_bar.numberOfOrbitals(OccupationType::k_virtual);
+        const auto& n_occ_sigma_bar = orbital_space_sigma_bar.numberOfOrbitals(OccupationType::k_occupied);
+        const auto& n_virt_sigma_bar = orbital_space_sigma_bar.numberOfOrbitals(OccupationType::k_virtual);
 
-        // We need the two-electron integrals in mixed MO basis.
-        // The ground state coefficient matrix is obtained from the QCModel.
-        const auto& two_electron_integrals = rsq_hamiltonian.twoElectron();
-
-        // We transform the necessary integrals to a 'MixedUSQTwoElectronOperatorComponent` in order to be able to transform the tensor indices to a mixed basis.
-        const auto g_op = GQCP::ScalarMixedUSQTwoElectronOperatorComponent<Scalar> {two_electron_integrals.parameters()};
-
-        // Transform the first two indices to the sigma basis and the last two indices to the sigma_bar basis.
-        // Return the parameters for later use.
-        const auto g = g_op.transformed(this->expansion(Spin::alpha), Spin::alpha).transformed(this->expansion(Spin::beta), Spin::beta).parameters();
+        // The mixed alpha-beta two electron integrals are extracted from the Hamiltonian.
+        const auto& g = usq_hamiltonian.twoElectron().alphaBeta().parameters();
```

Comment:
```suggestion
        const auto& g_aabb = usq_hamiltonian.twoElectron().alphaBeta().parameters();
```

For extra clarity, I think it's better to keep the `aabb` subscript, so we don't get confused which two-electron integrals we're using.

This comment also applies to the other methods, where you use `g_aaaa` and `g_bbbb`